### PR TITLE
Add a blog post for visualizing graphs with grafana

### PR DESCRIPTION
**Pull request by shana**
What the title says!

### CODE REVIEW

**Comment by shana** [comment]
Package is published at https://ab.bot/packages/aseriousbiz/grafana but it's marked as unlisted right now. Remember to list it when this goes up!

File: _posts/2021/2021-07-01-graphs-in-chat.markdown [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+title: "Visualize Grafana dashboards in chat"
+description: "The grafana skill lets you quickly see all your Grafana dashboards and visualize custom graph queries without leaving your chat window."
+tags: [grafana,graphs]
+excerpt_image: https://user-images.githubusercontent.com/310137/124177982-4369b600-dab1-11eb-84eb-33f3ba1d23c4.png
+author:
+    avatar: https://gravatar.com/avatar/dc6fa84a4a864f9627e220b716572cba?s=160
+    name: shana
+---
+
+An image is worth a thousand words, which is why Abbot can now help you visualize your graphs on chat with the [grafana](https://ab.bot/packages/aseriousbiz/grafana) skill!
```
Comment (haacked) [inline _posts/2021/2021-07-01-graphs-in-chat.markdown:11]:
```suggestion
An image is worth a thousand words, which is why [Abbot](https://ab.bot/) can now help you visualize your graphs in chat with the [grafana](https://ab.bot/packages/aseriousbiz/grafana) skill!
```

File: _posts/2021/2021-07-01-graphs-in-chat.markdown [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+List all your dashboards and quickly query your panels, generating graphs with custom time ranges and template variables to suit your visualization needs.
+
+## Setting it up
+Before you can start visualizing all your graphs, you need to tell Abbot where to go to get them.
+
+Start by generating a Grafana API key on your Grafana host, usually found at https://your-grafana-host/org/apikeys. Then, visit [https://ab.bot/skills/grafana](https://ab.bot/skills/grafana) and click "Manage skill secrets" to add a secret named ___GrafanaKey___, with the key you created.
```
Comment (haacked) [inline _posts/2021/2021-07-01-graphs-in-chat.markdown:18]: I think it looks good as code to indicate they should change it.

```suggestion
Start by generating a Grafana API key on your Grafana host, usually found at `https://your-grafana-host/org/apikeys`. Then, visit [https://ab.bot/skills/grafana](https://ab.bot/skills/grafana) and click "Manage skill secrets" to add a secret named ___GrafanaKey___, with the key you created.
```

File: _posts/2021/2021-07-01-graphs-in-chat.markdown [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+## Customize your graphs
+Grafana graphs support templated variables and a host of other customization options, and you can use all of those from chat as well - the skill supports custom queries with time ranges, template variables, timezone specifications and size constraints.
+
+Most often, you'll want to customize the time window of your graph. You can specify a range by adding time parameters to your query:
+
+```bash
+@abbot grafana db home:3 2d 1d
+```
+
+This generates a graph of the `logins` panel in the `Home` dashboard with a time window of 2 days to 1 day from now.
```
Comment (haacked) [inline _posts/2021/2021-07-01-graphs-in-chat.markdown:75]: I'm having trouble parsing what this time window means. Does it mean a time window of 2 days ending 1 day ago? Or a time window starting 2 days ago and ending 1 day ago?

File: _posts/2021/2021-07-01-graphs-in-chat.markdown [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+Time parameters are optional, and if you only have one, it will be interpreted as a window of "now - Time" to now.
+
+
+```bash
+@abbot grafana db "grafana dashboard:client side full page load" 24h var-app=backend var-server=backend_01 var-server=backend_03 var-interval=1h
+```
+
+A more complex query generated a templated Grafana graph with a time window of 24 hours, customized with several variables.
```
Comment (haacked) [inline _posts/2021/2021-07-01-graphs-in-chat.markdown:84]: Is this sentence connected to the previous command?

```suggestion
The more complex query above generates a templated Grafana graph with a time window of 24 hours, customized with several variables.
```

File: _posts/2021/2021-07-01-graphs-in-chat.markdown [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+
+![Screenshot showing a graph with custom ranges and template variables](https://user-images.githubusercontent.com/310137/124180188-28e50c00-dab4-11eb-9398-d3333323bae0.png)
+
+Any parameter that Grafana supports in a query can be passed to the skill as key=value pairs, like timezone specifications or custom graph sizes:
```
Comment (haacked) [inline _posts/2021/2021-07-01-graphs-in-chat.markdown:89]:
```suggestion
Any parameter that Grafana supports in a query can be passed to the skill as `key=value` pairs, like timezone specifications or custom graph sizes:
```

**Review by haacked — CHANGES_REQUESTED** [review]
This is fantastic! Just some minor suggestions and questions.

Could you copy the code of the `grafana` skill to a `grafana.cs` file in https://github.com/aseriousbiz/abbot-skills/tree/main/skills. And mention if anyone has issues or suggested improvements to the skill to submit it to that repository? I'll manually integrate changes as they come in (if they come in).

File: _posts/2021/2021-07-01-graphs-in-chat.markdown (hunk 1 above)
Comment (shana) [reply _posts/2021/2021-07-01-graphs-in-chat.markdown:11]: Oh yes, good point, was going to do that and forgot! :+1:

File: _posts/2021/2021-07-01-graphs-in-chat.markdown (hunk 3 above)
Comment (shana) [reply _posts/2021/2021-07-01-graphs-in-chat.markdown:75]: Starting 2 days ago and ending 1 day ago. I'll rephrase!

**Comment by shana** [comment]
> This is fantastic! Just some minor suggestions and questions.
> 
> Could you copy the code of the `grafana` skill to a `grafana.cs` file in https://github.com/aseriousbiz/abbot-skills/tree/main/skills. And mention if anyone has issues or suggested improvements to the skill to submit it to that repository? I'll manually integrate changes as they come in (if they come in).

Updated the blog post with a link to the skill in the skills repo :+1:

**Pull request merged**